### DISCUSSION TITLE
Weird watch script behavior

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#6056] [Help and troubleshooting]
I had today a strange experience with our build script:
1. At some point when I re-run it with different options, it stopped to show electron, it just showed information about finishing successfully. I haven't seen any obvious hints in logs. When I started the script again with different options, I still had the same problem. Unfortunately I cannot post logs here, because
2. When I re-run the script while capturing logs to file, the behavior changed: now the electron showed up properly, but the logs enter some infinite loop, printing repetitive messages:

[log.txt](https://github.com/enso-org/enso/files/11048634/log.txt)

I work in train today, so maybe it has something to do with unstable connectivity. But the issues appeared also when I had viable internet connection.

Still, I have no clue what happens. @mwu-tow any suggestions?

// Edit: I caught the non-appearing electron again, this time I stored logs:

[log4.txt](https://github.com/enso-org/enso/files/11051762/log4.txt)

**Comment by vitvakatu** [comment]
I think I had a similar issue to `1.`, reported here: https://github.com/enso-org/enso/issues/6012
It disappeared after the restart, but clean builds didn't help.

**Reply by farmaazon** [reply]
I tried restarting, but still have this infinite loop.

**Reply by wdanilo** [reply]
@vitvakatu try `export USE_CARGO_WATCH_PLUS=1` and check if it fixes the issue. It fixes the exact same issue on my end. (you have to install `cargo-watch-plus` from crates.io.

**Comment by mwu-tow** [comment]
@farmaazon 
2. This endless loop seems to be driven by cargo-watch. I remember that @wdanilo had a similar issue, ergo https://github.com/enso-org/cargo-watch-plus 
You can try and see if it solves the issue.

> // Edit: I caught the non-appearing electron again, this time I stored logs:
>
> [log4.txt](https://github.com/enso-org/enso/files/11051762/log4.txt)

This also looks like an infinite loop from cargo-watch.

**Reply by farmaazon** [reply]
> This also looks like an infinite loop from cargo-watch.

I also spotted the infinite loop, but this time electron also didn't show, so I hoped you find the solution for the issue number 1 in my original post.

Or does the loop make electron not launching?

**Reply by mwu-tow** [reply]
@farmaazon 
I am not sure. Apart from loop, it just gets stuck. Try getting rid of the loop and check if this helps. I don't see any other simple explanation.

**Reply by wdanilo** [reply]
@farmaazon try `export USE_CARGO_WATCH_PLUS=1` and check if it fixes the issue. It fixes the exact same issue on my end.
